cmd/openstack-populator/openstack-populator_test.go

```diff
@@ -0,0 +1,145 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+)
+
+func setupMockServer() (*httptest.Server, string, int, error) {
+	listener, err := net.Listen("tcp", ":0")
+	if err != nil {
+		return nil, "", 0, err
+	}
+
+	mux := http.NewServeMux()
+
+	port := listener.Addr().(*net.TCPAddr).Port
+	identityServerURL := fmt.Sprintf("http://localhost:%d", port)
+
+	mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		response := fmt.Sprintf(`{
+            "versions": {
+                "values": [
+                    {
+                        "id": "v3.0",
+                        "links": [
+                            {"rel": "self", "href": "%s/v3/"}
+                        ],
+                        "status": "stable"
+                    }
+                ]
+            }
+        }`, identityServerURL)
+		fmt.Fprint(w, response)
+	})
+
+	mux.HandleFunc("/v2/images/", func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintln(w, `mock_data`)
+	})
+
+	mux.HandleFunc("/v3/auth/tokens", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		w.Header().Set("X-Subject-Token", "MIIFvgY")
+		w.WriteHeader(http.StatusCreated)
+		response := `{
+            "token": {
+                "methods": ["password"],
+                "project": {
+                    "domain": {
+                        "id": "default",
+                        "name": "Default"
+                    },
+                    "id": "8538a3f13f9541b28c2620eb19065e45",
+                    "name": "admin"
+                },
+				"catalog": [
+				{
+					"type": "image",
+                    "name": "glance",
+							"endpoints": [{
+								"url": "http://localhost:%d/v2/images",
+								"region": "RegionOne",
+								"interface": "public",
+								"id": "29beb2f1567642eb810b042b6719ea88"
+
+							}]
+						}],
+                "user": {
+                    "domain": {
+                        "id": "default",
+                        "name": "Default"
+                    },
+                    "id": "3ec3164f750146be97f21559ee4d9c51",
+                    "name": "admin"
+                },
+                "issued_at": "201406-10T20:55:16.806027Z"
+            }
+        }`
+		response = fmt.Sprintf(response, port)
+		fmt.Fprint(w, response)
+	})
+
+	server := httptest.NewUnstartedServer(mux)
+	server.Listener = listener
+
+	server.Start()
+
+	return server, identityServerURL, port, nil
+}
+
+func TestPopulate(t *testing.T) {
```

Comment:
I see the test is being exercised but you probably want it integrated with ginkgo like the others https://github.com/kubevirt/containerized-data-importer/blob/0c15a1c96cefad360235cb56473240ce04b7d18b/cmd/cdi-cloner/clone-source_suite_test.go#L1-L13

Reply:
changed to ginkgo